### PR TITLE
Static Build fix/enhancements

### PR DESCRIPTION
**Pull request by arvindcheru**
Proposed Changes:
 - Few install() statement were not using rocm-cmake APIs, this is updated.
 - Static Build issue fix (INSTALL_RPATH properties setting exempted for static build)

### CODE REVIEW

**Review by wenkaidu — APPROVED** [review]

**Comment by wenkaidu** [comment]
@arvindcheru let's wait for ci:extended and gfx942 testing to complete before merging. These tests need to be manually triggered by adding these labels

**Comment by arvindcheru** [comment]
[like]  Cheruvally, Aravindan reacted to your message:
________________________________

**Pull request merged**
